src/Network/Response.php

```diff
@@ -538,13 +538,18 @@ protected function _sendHeader($name, $value = null)
 
     /**
      * Sends a content string to the client.
+     * If the content is a callable, it is invoked.
      *
      * @param string $content string to send as response body
      * @return void
      */
     protected function _sendContent($content)
     {
-        echo $content;
+        if (!is_string($content) && is_callable($content)) {
+            $content->__invoke();
```

Comment:
@reinink I understand use case. It's just that doing `echo call_user_func($content);` _allows_ you to return value from the callback. For your use case the statement it will just evaluate to `echo null;`.

That said personally I am fine the way it is.